File: khairul_anik/1480/runningSum.py
from typing import List

# Time complexity: O(n), Space complexity: O(n)
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        last_added = nums[0]
        result = [last_added]

        for i in range(1, len(nums)):
            new = last_added + nums[i]
            result.append(new)
            last_added = new

        return result

solution = Solution()
print(solution.runningSum([1, 2, 3, 4]))  # Output: [1, 3, 6, 10]

File: farhad/1480/running_Sum.js
const runningSum = (nums) => {
  let sum = 0;
  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    nums[i] = sum;
  }
  return nums;
};

const nums = [1, 2, 3, 4];
const result = runningSum(nums);
console.log(result); // Output: [1, 3, 6, 10]
